public/js/index.js

```diff
@@ -47,129 +48,88 @@
-        state: 'none',
-        ui: 'lightbox',
-        redirectUri: config.fxa.redirectUri,
-        // The seller would need to request an access token with these
-        // scopes for payments to work.
-        scope: 'profile:email payments',
-      })
-      .then(function (res) {
-        console.log('login succeeded:', res);
-        console.log('requesting token for code', res.code);
-        $.ajax({
-          type: 'post',
-          url: 'https://oauth-stable.dev.lcip.org/v1/token',
-          dataType: 'json',
-          data: {
-            code: res.code,
-            client_id: config.fxa.client_id,
-            client_secret: config.fxa.client_secret,
-          },
+          // If we set state to a random string and verify that it matches
+          // on the return then we'd get some CSRF protection. However,
+          // it may only apply to redirect returns, I'm not sure.
+          state: 'none',
+          ui: 'lightbox',
+          redirectUri: config.fxa.redirectUri,
+          // The seller would need to request an access token with these
+          // scopes for payments to work.
+          scope: 'profile:email payments',
         })
-        .then(function(result) {
-          console.log('token result:', result);
-          resolve(result);
+        .then(function(res) {
+          console.log('login succeeded:', res);
+          console.log('requesting token for code', res.code);
+          $.ajax({
+              type: 'post',
+              url: 'https://oauth-stable.dev.lcip.org/v1/token',
+              dataType: 'json',
+              data: {
+                code: res.code,
+                client_id: config.fxa.client_id,
+                client_secret: config.fxa.client_secret,
+              },
+            })
+            .then(function(result) {
+              console.log('token result:', result);
+              resolve(result);
+            }, function(err) {
+              console.error('token failure:', err.responseJSON);
+              reject(err);
+            });
+
         }, function(err) {
-          console.error('token failure:', err.responseJSON);
+          console.error('sign-in failure:', err);
           reject(err);
         });
-
-      }, function (err) {
-        console.error('sign-in failure:', err);
-        reject(err);
-      });
     });
   }
 
+  // This is where we call the purchase method. Since we have multiple
+  // buy buttons on this one page, we'll use a function to avoid
+  // copying and pasting the code.
+  $('button.brick').on('click', function(event) {
+    purchase({
+      'id': 'mozilla-concrete-brick',
+      'image': 'http://bit.ly/default-png'
+    });
+  });
+
+  $('button.mortar').on('click', function(event) {
+    purchase({
+      'id': 'mozilla-concrete-mortar',
+      'image': 'http://bit.ly/mortar-png'
+    });
+  });
+
+  $('button.management').on('click', function(event) {
+    window.location = config.manageUrl;
```

Comment:
This should log in first, then redirect to the management app with a token. The old code did this:
````
fxaSignIn()
  .then(function(result) {
    window.location.href = config.manageUrl + '?access_token=' + result.access_token;
  });
````
We were mainly just using this as a way to test token sign-ins to management since you can already test user/password sign-ins by directly hitting the management URL (i.e. without a token).